Competition2017/src/competition/subsystems/vision/VisionSubsystem.java

```diff
@@ -34,54 +36,51 @@
     private static final String targetSnapshotPacketType = "trackedTargetsSnapshot";
     private static final String trackedLiftPegsProperty = "trackedLiftPegs";
     private static final String trackedBoilersProperty = "trackedBoilers";
-    
+
+    private double lastTimePacketRecieved;
+    private OffboardCommPacket packet;
+
     @Inject
     public VisionSubsystem(WPIFactory factory, XPropertyManager propManager) {
         log.info("Creating VisionSubsystem");
 
         server.setNewPacketHandler(packet -> handleCommPacket(packet));
-        
+
         server.start();
         log.info("Started server");
+        lastTimePacketRecieved = 0;
     }
-    
+
     public DetectedLiftPeg getTrackedLiftPeg() {
-        return trackedLiftPegs.stream()
-                .filter(target -> target.isTracked)
-                .findFirst()
-                .orElse(null);
+        return trackedLiftPegs.stream().filter(target -> target.isTracked).findFirst().orElse(null);
     }
-    
+
     public DetectedBoiler getTrackedBoiler() {
-        return trackedBoilers.stream()
-                .filter(target -> target.isTracked)
-                .findFirst()
-                .orElse(null);
+        return trackedBoilers.stream().filter(target -> target.isTracked).findFirst().orElse(null);
     }
-    
+
     private void handleCommPacket(OffboardCommPacket packet) {
-        // TODO: Only log occasionally, and track disconnects
-        log.info(packet.id + " (" + packet.packetType + "): " + packet.payload.toString());
-        
-        
-        if(packet.packetType.equals(targetSnapshotPacketType)) {
+        lastTimePacketRecieved = Timer.getFPGATimestamp();
```

Comment:
Keep track of the "packets per second" value in a class variable, and update it here. You could do this many ways, but I'd recommend this:

- Add a class variable indicating the number of packets received since the last time the variable was cleared.
- Add a class variable indicating the last time the variable was cleared.
- Every time a packet is received, increment it.
- In the `updatePeriodicData` method, check if the time since that variable was last cleared is greater than the reset threshold... use a `DoubleProperty` for the time threshold (as mentioned below). If so:
  - Divide the number of packets received by the amount of time since it was cleared, and store that in a class field
  - Clear the number of packets received and update the clear time.

When you need to check the current "packets per second", read from that class field.